dotCMS/src/main/java/com/dotcms/publisher/util/DependencyManager.java

```diff
@@ -878,6 +889,12 @@ private void processTemplateDependencies(final String id) {
 					}
 				}
 			}
+
+			if(workingTemplate instanceof FileAssetTemplate){
+				//Process FileAssetTemplate
+				final Set<Folder> folders = this.collectFileAssetTemplateDependencies(FileAssetTemplate.class.cast(workingTemplate));
+				setFolderListDependencies(folders);
```

Comment:
really you not need to call  setFolderListDependencies here, just add the folders into the dependencyProcesor
```suggestion
				dependencyProcessor.putAll(folders, AssetTypes.FOLDER);
```

really the putAll method not exists but it is easy to implement

The dependencyProcesor call the setFolderListDependencies later